docs/ExtensionUsage.md

```diff
@@ -0,0 +1,542 @@
+
+	/**
+	 * Returns the version number of this schema.
+	 */
+	@Override
+	public String getSchemaVersion() { ... }
+
+	/**
+	 * Returns the unique schema identifier.
+	 */
+	@Override
+	public String getSchemaIdentifier() { ... }
+
+	/**
+	 * Returns the unique dataset identifier. When this value is set the default Adobe Experience Platform Experience dataset configured in your Edge Configuration is overwritten.
+	 */
+	@Override
+	public String getDatasetIdentifier() { ... }
+
+	@Override
+	public Map<String, Object> serializeToXdm() { 
+  	Map<String, Object> map = new HashMap<>();
+    if (this.eventType != null) {
+			map.put("eventType", this.eventType);
+		}
+    
+    if (this.otherField != null) {
+			map.put("otherField", this.otherField);
+		}
+  }
+  
+  ...
+  
+	public String getEventType() {
+		return this.eventType;
+	}
+  
+  public void setEventType(final String newValue) {
+		this.eventType = newValue;
+	}
+  
+  public String getOtherField() {
+		return this.otherField;
+	}
+  
+  public void setOtherField(final String newValue) {
+		this.otherField = newValue;
+	}
+}
+
+// Create Experience Event from Schema
+XDMSchemaExample xdmData = new XDMSchemaExample();
+xdmData.setEventType("SampleXDMEvent");
+xdmData.setOtherField("OtherFieldValue");
+
+ExperienceEvent experienceEvent = new ExperienceEvent.Builder()
+      															.setXdmSchema(xdmData)
+																		.build();
+```
+
+
+
+**Add free form data to the Experience event:**
+
+If you would like to add non-XDM formatted data to the Experience event, follow the examples below. This may be useful for data elements or rules use-cases.
+
+###### Swift
+
+```swift
+let experienceEvent = ExperienceEvent(xdm: xdmData, data: ["free": "form", "data": "example"])
+```
+
+###### Android
+
+```java
+Map<String, Object> data = new HashMap<>();
+data.put("free", "form");
+data.put("data", "example");
+ExperienceEvent experienceEvent = new ExperienceEvent.Builder()
+  .setXdmSchema(xdmData)
+  .setData(data)
+  .build();
+```
+
+
+
+**Set the destination Dataset identifier to the current Experience event:**
+
+If you would like to send an Experience event a specific Dataset in Adobe Experience Platform which is not the default Experience Event dataset configured in your Edge Configuration, follow the examples below:
+
+###### Swift
+
+```swift
+let experienceEvent = ExperienceEvent(xdm: xdmData, datasetIdentifier: "datasetIdExample")
+```
+
+###### Android
+
+```java
+ExperienceEvent experienceEvent = new ExperienceEvent.Builder()
+				.setXdmSchema(xdmData, "datasetIdExample")
+				.build();
+```
+
+
+
+**Send an Experience Event:**
+
+###### Swift
+
+```swift
+Edge.sendEvent(experienceEvent: experienceEvent)
+```
+
+###### Android
+
+```java
+Edge.sendEvent(experienceEvent, null);
+```
+
+By default this API sends the `experienceEvent` to Adobe Experience Edge Network to the Adobe Experience Platform Experience dataset configured in your Edge Configuration.
```

Comment:
Maybe reword to:
"The `sendEvent` API sends the `experienceEvent` to the Adobe Experience Edge Network. If a dataset ID is not set in the ExperienceEvent, then by default the `experienceEvent` is sent to the dataset configured in the application's Edge Configuration."